cpp/tests/algo_factory.cc

```diff
@@ -109,7 +109,7 @@ TEST_CASE("primal_dual_factory") {
   auto const primaldual =
       factory::primaldual_factory<sopt::algorithm::ImagingPrimalDual<t_complex>>(
           factory::algo_distribution::serial, measurements_transform, wavelets, uv_data, sigma,
-          imsizey, imsizex, sara.size(), 1000, true, true, 1e-2, 1, op_norm);
+          imsizey, imsizex, sara.size(), 20, true, true, 1e-2, 1, op_norm);
 
   auto const diagnostic = (*primaldual)();
   CHECK(diagnostic.niters == 16);
```

Comment:
Doesn't really matter since we expect it to fail, but I would still remove this check